zanata-war/src/main/java/org/zanata/service/impl/WebHooksPublisher.java

```diff
@@ -22,37 +22,69 @@
 package org.zanata.service.impl;
 
 import javax.annotation.Nonnull;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
 import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
 
-import org.jboss.resteasy.client.ClientRequest;
-import org.jboss.resteasy.client.ClientResponse;
-import org.zanata.events.JSONType;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
+import org.zanata.events.WebhookEventType;
 
+import com.google.common.base.Optional;
 import lombok.extern.slf4j.Slf4j;
+import org.zanata.util.HmacUtil;
 
 /**
+ * Do http post for webhook event
+ *
  * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
  */
 @Slf4j
 public class WebHooksPublisher {
 
-    private static ClientResponse publish(@Nonnull String url,
+    public static final String WEBHOOK_HEADER = "X-Zanata-Webhook";
+
+    public static Response publish(@Nonnull String url,
+        @Nonnull WebhookEventType event, Optional<String> secretKey,
+        @Nonnull String serverUrl) {
+        return publish(url, event.getJSON(), MediaType.APPLICATION_JSON_TYPE,
+            MediaType.APPLICATION_JSON_TYPE, serverUrl, secretKey);
+    }
+
+    private static Response publish(@Nonnull String url,
             @Nonnull String data, @Nonnull MediaType acceptType,
-            @Nonnull MediaType mediaType) {
+            @Nonnull MediaType mediaType, String serverUrl,
+            Optional<String> secretKey) {
         try {
-            ClientRequest request = new ClientRequest(url);
-            request.accept(acceptType);
-            request.body(mediaType, data);
-            return request.post();
+            ResteasyClient client = new ResteasyClientBuilder().build();
+            ResteasyWebTarget target = client.target(url);
+
+            Invocation.Builder postBuilder =
+                    target.request().accept(acceptType);
+
+            String sha = signWebhookHeader(data, secretKey.get(), serverUrl);
+            if (secretKey.isPresent()) {
+                postBuilder.header(WEBHOOK_HEADER, sha);
+            }
```

Comment:
Won't line 67 blow up if the secretKey is not present?  In any case, you don't need to generate the sha unless the secret is present.